ports/libxml2/vcpkg-cmake-wrapper.cmake

```diff
@@ -1,77 +1,7 @@
-cmake_policy(PUSH)
-cmake_policy(SET CMP0012 NEW)
-cmake_policy(SET CMP0054 NEW)
-if(POLICY CMP0079)
-    cmake_policy(SET CMP0079 NEW)
-endif()
-list(FIND ARGS "CONFIG" Z_VCPKG_LIBXML2_CONFIG)
-
-set(LIBXML2_INCLUDE_DIR "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include" CACHE PATH "" FORCE)
-find_library(LIBXML2_LIBRARY_DEBUG
-    NAMES xml2 libxml2 xml2s libxml2s xml2d libxml2d xml2sd libxml2sd
-    NAMES_PER_DIR
-    PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib"
-    NO_DEFAULT_PATH
-)
-find_library(LIBXML2_LIBRARY_RELEASE
-    NAMES xml2 libxml2 xml2s libxml2s
-    NAMES_PER_DIR
-    PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib"
-    NO_DEFAULT_PATH
-)
-include(SelectLibraryConfigurations)
-select_library_configurations(LIBXML2)
-unset(LIBXML2_FOUND)
-if(CMAKE_VERSION VERSION_LESS "3.10")
-    set(LIBXML2_INCLUDE_DIRS "${LIBXML2_INCLUDE_DIR}")
-    set(LIBXML2_LIBRARIES "${LIBXML2_LIBRARIES}" CACHE STRING "CMake 3.10 cache variable" FORCE)
-else()
-    set(LIBXML2_LIBRARY "${LIBXML2_LIBRARIES}" CACHE STRING "CMake 3.10 cache variable" FORCE)
-endif()
+set(z_vcpkg_libxml2_module_path "${CMAKE_MODULES_PATH}")
+set(CMAKE_MODULES_PATH "${CMAKE_MODULES_PATH};${CMAKE_CURRENT_LIST_DIR}")

```

Comment:
What is the purpose of the redirection via another find module? Let me guess: 
- to ensure Find module mode
- to use the find module interface variables 

instead of treating `ARGS` like in https://github.com/dg0yt/vcpkg/blob/e2794913924ef606cc7bbc59c173f2714e9853c5/ports/curl/vcpkg-cmake-wrapper.cmake#L10-L11

Probably it is a good idea. At least it is the only reliable way to tell Find module mode from config mode. Config mode can be triggered by additonal keywords.

But AFAICT this would be the first port to implement the pattern. It is another indirection and file, and I'm not convinced we understand all the side effects now. I also wonder if this redirection shouldn't start in `find_package`.

So before adding a new practice, I would prefer to stay with the one found in `curl` and other ports.

Reply:
Now I see the remark in https://github.com/microsoft/vcpkg/pull/24935#discussion_r890566870
> I also add this override only onto the end of the CMAKE_MODULE_PATH, so it will not replace a user's custom copy; it should only get hit if they would have gotten the built-in module.

I think I would still prefer to take the simpler approach, in line with existing wrapper patterns.